swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -492,3 +492,15 @@ extension Adapter: CallbackHandlerDelegate {
     }
   }
 #endif
+
+extension Network.NWPath {
+
+  // We define a path as different from another if the following properties change
+  func connectivityDifferentFrom(path: Network.NWPath?) -> Bool {
+    guard let path = path else { return true }
```

Comment:
So if the new path is null, we consider that different and do a reset? What is the point of that?

Reply:
Yeah, we need to initialize it somehow. There is a (highly unlikely) chance we can receive this callback, `lastRelevantPath` is nil, and we need to bump sockets so better to err on the side of one reset.

Reply:
You could initialise it (and call reset) without calling this function. That may be easier to understand how we handle that edge-case?

Reply:
Updated!